### ISSUE TITLE
Compilation fails on some versions of OS X (`abs`)

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4049]
Noticed by @larkost on his computer:
```
src/rdb_protocol/datum.cc:618:17: error: using integer absolute value function 'abs' when argument is of floating point type [-Werror,-Wabsolute-value]
        value = abs(value);
                ^
src/rdb_protocol/datum.cc:618:17: note: use function 'std::abs' instead
        value = abs(value);
                ^~~
                std::abs
src/rdb_protocol/datum.cc:618:17: note: include the header <cmath> or explicitly provide a declaration for 'std::abs'
1 error generated.
```

I cannot reproduce this on gantz, so it probably depends on the exact environment and versions.

@Tryneus is looking into it.

**Comment by AtnNn** [comment]
I remember seeing this before, I think I had worked around it by using `fabs` instead.

**Comment by Tryneus** [comment]
Could not reproduce this, so fixed in `next` in commit 2d1e38a, and cherry-picked into `v2.0.x` in commit ed476b4, using what the compiler error suggested.  Will be in release 2.0.  @larkost, please reopen if this problem persists.
